host-services/builtins/builtins_client.go

```diff
@@ -43,11 +47,15 @@ func (c *BuiltinServicesClient) KVGet(key string) ([]byte, error) {
 	return resp.Data, nil
 }
 
-func (c *BuiltinServicesClient) KVSet(key string, value []byte) (*agentapi.HostServicesKeyValueResponse, error) {
+func (c *BuiltinServicesClient) KVSet(key string, value []byte, traceID *string) (*agentapi.HostServicesKeyValueResponse, error) {
 	metadata := map[string]string{
 		agentapi.KeyValueKeyHeader: key,
 	}
 
+	if traceID != nil {
+		metadata[agentapi.OtelTraceparent] = *traceID
```

Comment:
again, this belongs in a context which can/should then be passed to the builtins client